### PR TITLE
Implemented Serialization and TwoHandNavigator

### PR DESCRIPTION
**Pull request by rabaraaaron**
Android: Setup new project for Motor Control
Android: Build test app
Android: Serialization and Deserialization for Tremor, Kinetic Tremor, and Tapping
Android: Navigation for Two Hand Assessments

### CODE REVIEW

File: MotorControl/src/main/kotlin/org/sagebionetworks/motorcontrol/navigation/TwoHandNavigator.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    private val nodes: List<Node>
+    private val handSelectionIdentifier = "handSelection"
+
+    init {
+        if(node.children.toSet().size != node.children.size) {
+            throw IdentifiersNotUniqueException("Identifiers are not unique")
+        }
+        val swapHands: Boolean = (0..1).random() == 1
+        val tempNodesList = node.children.toMutableList()
+        val firstHandIndex = tempNodesList.indexOfFirst { it.hand() != null }
```
Comment (rabaraaaron) [inline MotorControl/src/main/kotlin/org/sagebionetworks/motorcontrol/navigation/TwoHandNavigator.kt:52]: Add comment here about assuming the hands are next to each other in step sequence

File: MotorControl/src/main/kotlin/org/sagebionetworks/motorcontrol/navigation/TwoHandNavigator.kt [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    private fun previousNode(currentNode: Node?): Node? {
+        if(currentNode == null || isCompleted(currentNode)) {
+            return null
+        }
+        val currentNodeIndex = nodes.indexOfFirst { it.identifier == currentNode.identifier }
+        if(currentNodeIndex < 1) {
+            return null
+        }
+        return nodes[currentNodeIndex - 1]
```
Comment (rabaraaaron) [inline MotorControl/src/main/kotlin/org/sagebionetworks/motorcontrol/navigation/TwoHandNavigator.kt:135]: Needs to account for navigating back on a section step and skipping hands we don't want to navigate back to.

File: MotorControl/src/main/kotlin/org/sagebionetworks/motorcontrol/serialization/MotorControlNodes.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import org.sagebionetworks.assessmentmodel.serialization.StepObject
+import org.sagebionetworks.motorcontrol.navigation.TwoHandNavigator
+
+val motorControlNodeSerializersModule = SerializersModule {
+    polymorphic(Node::class) {
+        subclass(TwoHandAssessmentObject::class)
+        subclass(HandInstructionStepObject::class)
+        subclass(TremorStepObject::class)
+        subclass(TappingStepObject::class)
+    }
```
Comment (rabaraaaron) [inline MotorControl/src/main/kotlin/org/sagebionetworks/motorcontrol/serialization/MotorControlNodes.kt:22]: Add TwoHandAssessmentObject under polymorphic(AssessmentObject::class)

File: app/src/test/kotlin/TwoHandNavigationTests.kt [hunk 4]
```diff
@@ -0,0 +1,331 @@
+import kotlinx.datetime.Clock
```
Comment (rabaraaaron) [inline app/src/test/kotlin/TwoHandNavigationTests.kt:1]: Move unit test code to the MotorControl module

File: app/src/test/kotlin/TwoHandNavigationTests.kt [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+
+    @Test
+    fun testTwoHandForwardNavigation_SectionNodeBefore() {
+        val assessmentObject = TwoHandAssessmentObject("foo", nodeList)
+        val nodeState = BranchNodeStateImpl(assessmentObject)
+        val navigator = assessmentObject.createNavigator(nodeState)
+        assertNotNull(navigator)
+        assertTrue(navigator is TwoHandNavigator)
+
+        val point = navigator.nodeBefore(nodeC, assessmentObject.createResult())
```
Comment (rabaraaaron) [inline app/src/test/kotlin/TwoHandNavigationTests.kt:174]: Will need to adjust after changing navigator's previous method

**Review by nategbrown9 — APPROVED** [review]
Looks good.

**Pull request merged**